File: content-planner-motia/steps/api.step.ts
import { ApiRouteConfig, Handlers } from 'motia'
import { z } from 'zod'

export const config: ApiRouteConfig = {
  type: 'api',
  name: 'ContentGenerationAPI',
  description: 'Triggers content generation from article URL',
  path: '/generate-content',
  method: 'POST',
  bodySchema: z.object({
    url: z.string().url({ message: 'Valid URL is required' })
  }),
  responseSchema: {
    200: z.object({
      message: z.string(),
      requestId: z.string(),
      url: z.string(),
      status: z.string()
    }),
    400: z.object({
      error: z.string()
    })
  },
  emits: ['scrape-article'],
  flows: ['content-generation']
}

export const handler: Handlers['ContentGenerationAPI'] = async (req, { emit, traceId }) => {
  const { url } = req.body

  await emit({
    topic: 'scrape-article',
    data: {
      requestId: traceId,
      url,
      timestamp: Date.now()
    }
  })

  return {
    status: 200,
    body: {
      message: 'Content generation started',
      requestId: traceId,
      url,
      status: 'processing'
    }
  }
}

File: content-planner-motia/steps/api.step.py
from pydantic import BaseModel, HttpUrl
from typing import Dict, Any

class RequestBody(BaseModel):
    url: HttpUrl

class SuccessResponse(BaseModel):
    message: str
    requestId: str
    url: str
    status: str

class ErrorResponse(BaseModel):
    error: str

config = {
    'type': 'api',
    'name': 'ContentGenerationAPI',
    'description': 'Triggers content generation from article URL',
    'path': '/generate-content',
    'method': 'POST',
    'bodySchema': RequestBody,
    'responseSchema': {
        200: SuccessResponse,
        400: ErrorResponse
    },
    'emits': ['scrape-article'],
    'flows': ['content-generation']
}

async def handler(input: Dict[str, Any], context):
    logger = context.logger
    state = context.state
    emit = context.emit
    trace_id = context.traceId
    
    # Extract request data
    req = input['req']
    url = req['body']['url']
    
    await emit({
        'topic': 'scrape-article',
        'data': {
            'requestId': trace_id,
            'url': str(url),
            'timestamp': int(__import__('time').time() * 1000)
        }
    })
    
    return {
        'status': 200,
        'body': {
            'message': 'Content generation started',
            'requestId': trace_id,
            'url': str(url),
            'status': 'processing'
        }
    }

